gammapy/analysis/tests/test_analysis.py

```diff
@@ -14,127 +14,86 @@
-            filters:
-            - filter_type: all
-            - filter_type: ids
-              exclude: true
-              obs_ids:
-              - 110380
-        result: 3
-      - observations:
-            datastore: $GAMMAPY_DATA/cta-1dc/index/gps/
-            filters:
-            - filter_type: sky_circle
-              frame: galactic
-              lat: 0 deg
-              lon: 0 deg
-              border: 0.5 deg
-              radius: 1 deg
-        result: 1
-      - observations:
-            datastore: $GAMMAPY_DATA/cta-1dc/index/gps/
-            filters:
-            - filter_type: angle_box
-              value_range:
-              - 265 deg
-              - 268 deg
-              variable: RA_PNT
-        result: 2
-      - observations:
-            datastore: $GAMMAPY_DATA/cta-1dc/index/gps/
-            filters:
-            - filter_type: par_box
-              value_range:
-              - 106000
-              - 107000
-              variable: EVENT_COUNT
-        result: 2
-      - observations:
-            datastore: $GAMMAPY_DATA/hess-dl3-dr1
-            filters:
-            - filter_type: par_value
-              value_param: Off data
-              variable: TARGET_NAME
-        result: 45
+def test_update():
+    cfg = AnalysisConfig()
+    analysis = Analysis(cfg)
+    data = {"general": {"outdir": "test"}}
+    analysis.update_config(data)
+    assert analysis.config.general.outdir == "test"
+
+    analysis = Analysis(cfg)
+    cfg2 = AnalysisConfig(**data)
+    analysis.update_config(cfg2)
+    assert analysis.config.general.outdir == "test"
+
+    analysis = Analysis(cfg)
+    data = """
+    general:
+        outdir: test
     """
-    config_obs = yaml.safe_load(cfg)
-    return config_obs
+    analysis.update_config(data)
+    assert analysis.config.general.outdir == "test"
+
+
+def test_config_to_yaml(tmp_path):
+    filename = "temp.yaml"
+    config = AnalysisConfig()
+    config.general.outdir = str(tmp_path)
+    config.to_yaml(filename=filename)
+    text = (tmp_path / filename).read_text()
+    assert "stack" in text
 
 
 @requires_data()
-@pytest.mark.parametrize("config_obs", config_observations())
-def test_get_observations(config_obs):
+def test_get_observations():
     config = AnalysisConfig()
     analysis = Analysis(config)
-    analysis.config.update_settings(config_obs)
+    analysis.config.data.datastore = "$GAMMAPY_DATA/cta-1dc/index/gps/"
+    analysis.get_observations()
+    assert len(analysis.observations) == 4
+    analysis.config.data.obs_ids = ["110380"]
+    analysis.get_observations()
+    assert len(analysis.observations) == 1
+    config = AnalysisConfig.from_template("1d")
+    analysis = Analysis(config)
     analysis.get_observations()
-    assert len(analysis.observations) == config_obs["result"]
+    assert len(analysis.observations) == 4
+    # TODO
+    # obs_file
+    # obs_time
 
 
 @pytest.fixture(scope="session")
 def config_analysis_data():
```

Comment:
This `config_analysis_data` is only used in a single test.
Remove the fixture and move it to that test.

Reply:
Done.

Reply:
Done.